eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -125,7 +125,9 @@ case class ChannelParams(channelId: ByteVector32,
     // README: if we set our bitcoin node to generate taproot addresses and our peer does not support option_shutdown_anysegwit, we will not be able to mutual-close
     // channels as the isValidFinalScriptPubkey() check would fail.
     val allowAnySegwit = Features.canUseFeature(localParams.initFeatures, remoteParams.initFeatures, Features.ShutdownAnySegwit)
-    if (localParams.upfrontShutdownScript_opt.exists(_ != localScriptPubKey)) Left(InvalidFinalScript(channelId))
+    val mustUseUpfrontShutdownScript = channelFeatures.hasFeature(Features.UpfrontShutdownScript)
```

Comment:
How can that work, it doesn't seem like `channelFeatures` ever contains `UpfrontShutdownScript`...
So this will always return `false`, won't it, even for channels that actually negotiated `upfront_shutdown_script` initially?

Reply:
> it doesn't seem like channelFeatures ever contains UpfrontShutdownScript

It does, why would it not? `UpfrontShutdownScript` is a `PermanentChannelFeatures`.

Reply:
You're right, I wasn't looking at the right place.